pkg/kubeclient/client.go

```diff
@@ -0,0 +1,89 @@
+package kubeclient
+
+import (
+	"context"
+
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/watch"
+)
+
+// Interface defines capability of the Kubernetes Client.
+type Interface interface {
+	// Create creates given object in Kubernetes cluster
+	Create(ctx context.Context, obj runtime.Object) error
+
+	// Get retrieves given object from Kubernetes cluster
+	Get(ctx context.Context, obj runtime.Object) error
```

Comment:
"retrieves given object". Could you be more specific? The hidden question here is, why do I need an object to get something?

Reply:
One of the thing which is sort of hidden is that we will be using methods defined in runtime.Object to determine REST endpoint for the resource involved. We could take those as input explicitly also, but then there will be duplication and verbosity in the caller's code.